pkg/prom/instance/instance.go

```diff
@@ -261,10 +263,16 @@ func New(reg prometheus.Registerer, cfg Config, walDir string, logger log.Logger
 func newInstance(cfg Config, reg prometheus.Registerer, logger log.Logger, newWal walStorageFactory) (*Instance, error) {
 	vc := NewMetricValueCollector(prometheus.DefaultGatherer, remoteWriteMetricName)
 
+	hostname, err := Hostname()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get hostname: %w", err)
+	}
+
 	i := &Instance{
-		cfg:    cfg,
-		logger: logger,
-		vc:     vc,
+		cfg:        cfg,
+		logger:     logger,
+		vc:         vc,
+		hostFilter: NewHostFilter(hostname, cfg.HostFilterRelabelConfigs),
```

Comment:
I moved the host filter to span the lifetime of the instance. This lets us update the host filter relabel configs, which wasn't working before (and wasn't treated as an invalid update, either).